File: Stack.java
class Stack {
    int[] arr;
    int top;
    static int Max = 100;

    Stack() {
        arr=new int[Max];
        top = -1;
    }

    void push(int val) {
        if (top == Max - 1) {
            System.out.println("Stack overflow");
        } else {
            arr[++top] = val;
        }
    }


    int pop() {
        if (top == -1) {
            System.out.println("Stack underflow");
            return 0;
        } else {
            int x = arr[top--];
            return x;
        }
    }

    boolean isEmpty() {
        return top < 0;
    }

    void display() {
        while (!isEmpty()) {
            System.out.println(arr[top] + "\t");
            top--;
        }
    }

    public static void main(String[] args) {
        Stack s1=new Stack();
        s1.push(10);
        s1.push(20);
        s1.push(30);
        s1.push(40);
        s1.pop();
        s1.display();
    }
}

File: stack.c
//created by akashbhalotia
#include<stdio.h>
#include<stdlib.h>

int isEmpty;

struct stack
{
	int data;
	struct stack *next;
} *top;

void push(int data)
{
	struct stack *newNode=(struct stack*)malloc(sizeof(struct stack*));
	isEmpty=0;
	
	newNode->data=data;
	newNode->next=top;
	top=newNode;
	
	printf("%d inserted.\n",data);
}
int pop()
{
	int data;
	if(top==NULL)
	{
		printf("Stack underflow!\n");
		data=-1;
		isEmpty=1;	
	}
	else
	{
		struct stack *tmp=top;
		top=top->next;
		data=tmp->data;
		
		free(tmp);
	}
	
	return data;
}
void printStack()
{
	struct stack *tmp=top;
	
	if(top==NULL)
		printf("Stack is empty.");
	while(tmp!=NULL)
	{
		printf("%d->",tmp->data);
		tmp=tmp->next;
	}
	printf("\n");
}
int main()
{
	int choice, data;
	top=NULL;
	
	isEmpty=1;
	
	do
	{
		printf("\nMENU:\n");
		printf("1)Push\n2)Pop\n3)Print Stack\n4)Exit\n");
		printf("Enter your choice.\n");
		
		scanf("%d",&choice);
		
		switch(choice)
		{
			case 1: 
			printf("Enter a number\n");
			scanf("%d",&data);
			push(data);
			break;
			
			case 2:
			data=pop();
			if(!isEmpty)
				printf("%d deleted.\n",data);
			break;
			
			case 3:
			printStack();
			break;
			
			case 4:
			printf("Goodbye!\n");
			break;
			
			default:
			printf("Invalid choice!\n");
		}
	}
	while(choice!=4);
	
	return 0;
}
